### PR TITLE
feature/185

### PR DESCRIPTION
**Pull request by hobowithshotgun**
Core logic and logs

### CODE REVIEW

File: HW6/Roman-Zuev/Trading/TradingCore.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            this.inputOutputDevice = inputOutputDevice;
+            this.phraseProvider = phraseProvider;
+            this.transactionHistoryRecorder = transactionHistoryRecorder;
+            this.transactionGenerator = transactionGenerator;
+            this.settings = settings;
+        }
+
+        private bool Transaction(int sellerId, int buyerId, int sharesId, int quantity, out DateTime TransactionDateTime)
+        {
+            using (TradingDBContext db = new TradingDBContext())
```
Comment (chesdenis) [inline HW6/Roman-Zuev/Trading/TradingCore.cs]: Real implementation of DbContext. Should be isolated.

**Comment by hobowithshotgun** [comment]
trading

scheme for the solution

File: HW6/Roman-Zuev/Trading/DataModel/Client.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+//using System.Collections.Generic;
+//using System.ComponentModel.DataAnnotations;
+//using System.ComponentModel.DataAnnotations.Schema;
+//using System.Linq;
+//using System.Text;
+//using System.Threading.Tasks;
+
+//namespace Trading.DataModel
+//{
+//    public class Client
```
Comment (chesdenis) [inline HW6/Roman-Zuev/Trading/DataModel/Client.cs]: commented code

File: HW6/Roman-Zuev/Trading/DataModel/ClientFund.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        [MaxLength(100)]
+        public string Name { get; set; }
+        [MaxLength(30)]
+        public string PhoneNumber { get; set; }
+        [Required]
+        public decimal Balance { get; set; }
+        [Required, MaxLength(10)]
+        public string Zone { get; set; }
+        public ICollection<ClientShares> ClientShares { get; set; }
+        public Client()
```
Comment (chesdenis) [inline HW6/Roman-Zuev/Trading/DataModel/ClientFund.cs]: what is the purpose of constructor here? if this is one to many - it should be virtual for ClientShares

File: HW6/Roman-Zuev/Trading/DataModel/TradingDBContext.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            Database.SetInitializer(new TradingDBInitializer());
+        }
+        public virtual DbSet<Client> Clients { get; set; }
+        //public virtual DbSet<ClientFund> ClientFunds { get; set; }
+        public virtual DbSet<ClientShares> ClientShares { get; set; }
+        public virtual DbSet<Shares> Shares { get; set; }
+        public virtual DbSet<TransactionHistory> TransactionHistories { get; set; }
+
+        //protected override void OnModelCreating(DbModelBuilder modelBuilder)
```
Comment (chesdenis) [inline HW6/Roman-Zuev/Trading/DataModel/TradingDBContext.cs]: commented code

**Pull request merged**